### PR TITLE
Implement the BooleanFilter

### PR DESCRIPTION
**Pull request by chervaliery**
In order to make query on boolean type, I implement the BooleanFilter.

### CODE REVIEW

File: Filter/BooleanFilter.php [hunk 1]
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                     continue;
                 }
 
-                $values[] = ($v == BooleanType::TYPE_YES) ? 1 : 0;
+                $values[] = ($v == BooleanType::TYPE_YES) ? true : false;
```
Comment (greg0ire) [inline Filter/BooleanFilter.php]: Could you change that to simply : `$values[] = ($v == BooleanType::TYPE_YES)` ? It gives the same result.

File: Filter/BooleanFilter.php [hunk 2]
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
...
 
             if (count($values) == 0) {
                 return;
             }
+            
+            $queryBuilder = new \Elastica\Query\Builder();
+            $queryBuilder
+            ->fieldOpen('terms')
+            ->field($field, $values)
+            ->fieldClose();
```
Comment (greg0ire) [inline Filter/BooleanFilter.php]: Can you indent these three lines ?

File: Filter/BooleanFilter.php [hunk 3]
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
...
             }
 
-            $parameterName = $this->getNewParameterName($queryBuilder);
-            $this->applyWhere($queryBuilder, sprintf('%s.%s = :%s', $alias, $field, $parameterName));
-            $queryBuilder->setParameter($parameterName, ($data['value'] == BooleanType::TYPE_YES) ? 1 : 0);
+            $queryBuilder = new \Elastica\Query\Builder();
+            $queryBuilder
+            ->fieldOpen('term')
+            ->field($field, ($data['value'] == BooleanType::TYPE_YES) ? true : false)
+            ->fieldClose();
```
Comment (greg0ire) [inline Filter/BooleanFilter.php]: Indention please

File: Filter/BooleanFilter.php [hunk 4]
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
...
                 return;
             }
 
-            $parameterName = $this->getNewParameterName($queryBuilder);
-            $this->applyWhere($queryBuilder, sprintf('%s.%s = :%s', $alias, $field, $parameterName));
-            $queryBuilder->setParameter($parameterName, ($data['value'] == BooleanType::TYPE_YES) ? 1 : 0);
+            $queryBuilder = new \Elastica\Query\Builder();
+            $queryBuilder
+            ->fieldOpen('term')
+            ->field($field, ($data['value'] == BooleanType::TYPE_YES) ? true : false)
```
Comment (greg0ire) [inline Filter/BooleanFilter.php]: Please remove this ternary.

**Comment by greg0ire** [comment]
Could you please write some tests ? #7 introduces one for the `StringFilter`

**Comment by greg0ire** [comment]
Your commit message is great! If you can fix all the things I pointed out, this will be merged quickly. Thanks a lot!

File: Filter/BooleanFilter.php (hunk 1 above)
Comment (chervaliery) [inline Filter/BooleanFilter.php]: It is not the same, 
BooleanType::TYPE_YES = 1 and BooleanType::TYPE_YES = 2
https://github.com/sonata-project/SonataCoreBundle/blob/master/Form/Type/BooleanType.php

Comment (greg0ire) [inline Filter/BooleanFilter.php]: `($v == BooleanType::TYPE_YES)` is a condition. It always returns a boolean (either `true` or `false`). This line is basically saying : If the condition is `true`, store `true`, if it is `false` store `false`. I'm saying we could replace that with just "store the result of the condition"

**Comment by chervaliery** [comment]
Is there still a problem? Do I need to change something else?

**Comment by greg0ire** [comment]
Huh yes there is a problem : there are 3 commits instead of one O_o

File: Tests/Filter/BooleanFilterTest.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		$queryProperty   = $queryReflection->getProperty('query');
+
+		$queryProperty->setAccessible(true);
+
+		$queryArray = $queryProperty->getValue($this->proxyQuery)->toArray();
+		 
+		$this->assertEquals('true', $queryArray['query']['bool']['must'][0]['term']['foo']);
+	}
+
+}
```
Comment (greg0ire) [inline Tests/Filter/BooleanFilterTest.php]: Please don't use tabs. Make sure you don't violate PSR. Try to integrate [this](http://cs.sensiolabs.org/) in your IDE if you can.

**Comment by greg0ire** [comment]
I noticed that the commit have the same subject, but not the same body. The commit subject is not a category and the body a category. I think that in fact, you can keep the 3 commits, but they should be reworded.

**Comment by greg0ire** [comment]
The first one is good.

**Comment by greg0ire** [comment]
The second one should read like this : 

> add tests for the boolean filter

but it could also be squashed with the first commit, your choice.

**Comment by greg0ire** [comment]
The third should read 

> remove unnecessary ternary
> 
> We want boolean here, not integers.

It must be squashed with the first commit if and the first commit does not work without it, otherwise you can keep it apart.

**Comment by chervaliery** [comment]
Ok, I will squash them and change the tabs

**Comment by greg0ire** [comment]
> Ok, I will squash them

I think this will be more simple :wink:

**Comment by chervaliery** [comment]
This time it should be good !

File: Resources/config/services.xml [hunk 6]
```diff
@@ -44,5 +44,10 @@
             class="Sonata\AdminSearchBundle\Filter\ChoiceFilter">
             <tag name="sonata.admin.filter.type" alias="elastica_choice" />
         </service>
+        <service
+            id="sonata.admin.search.filter.type.boolean"
+            class="Sonata\AdminSearchBundle\Filter\BooleanFilter">
+            <tag name="sonata.admin.filter.type" alias="elastica_boolean" />
```
Comment (greg0ire) [inline Resources/config/services.xml]: Can you rebase on master and realias this service appropriately ?

Comment (chervaliery) [inline Resources/config/services.xml]: I did it

Comment (greg0ire) [inline Resources/config/services.xml]: Nice! Let's wait for the build to finish.

**Comment by greg0ire** [comment]
Merged! Thanks a lot!

**Pull request merged**